src/main/java/com/kainos/ea/controller/WebService.java

```diff
@@ -38,6 +61,125 @@ public Response getJobRoles()
 
     }
 
+    @GET
+    @Path("/job-roles/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobRoleById(@PathParam("id") int id){
+        try {
+            return Response.status(HttpStatus.OK_200).entity(rolesService.getRoleById(id)).build();
+        } catch (Exception | DatabaseConnectionException  e){
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        } catch (JobRoleDoesNotExistException e){
+            return Response.status(HttpStatus.NOT_FOUND_404, e.getMessage()).build();
+        }
+    }
+
+    @PUT
+    @Path("/job-roles")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateJobRole(JobRoleXL jobRoleXL){
+        try{
+            if (jobRoleValidator.isValidJobRole(jobRoleXL)){
+                try {
+                    int id = rolesService.updateJobRole(jobRoleXL);
+                    return Response.status(HttpStatus.OK_200).entity(id).build();
+                } catch (Exception | DatabaseConnectionException e) {
+                    return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+                }
+            } else {
+                return Response.status(HttpStatus.BAD_REQUEST_400).build();
+            }
+        } catch (LinkSyntaxException e) {
+            return Response.status(HttpStatus.BAD_REQUEST_400).build();
+        }
+    }
+
+    @GET
+    @Path("/bands/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getBandById(@PathParam("id") int id){
+        try {
+            return Response.status(HttpStatus.OK_200).entity(bandsService.getBandById(id)).build();
+        } catch (Exception | DatabaseConnectionException  e){
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        } catch (BandDoesNotExistException e){
+            return Response.status(HttpStatus.NOT_FOUND_404, e.getMessage()).build();
+        }
+    }
```

Comment:
not sure why this endpoint is needed for your story?

Reply:
removed